lib_nbgl/src/nbgl_layout.c

```diff
@@ -3024,6 +2903,275 @@ int nbgl_layoutAddExtendedFooter(nbgl_layout_t *layout, const nbgl_layoutFooter_
     return layoutInt->footerContainer->obj.area.height;
 }
 
+/**
+ * @brief Creates a touchable area on top of the footer of the screen, containing various controls,
+ * described in the given structure. This up-footer is not part of the main container
+ *
+ * @param layout the current layout
+ * @param upFooterDesc description of the up-footer
+ * @return height of the control if OK
+ */
+int nbgl_layoutAddUpFooter(nbgl_layout_t *layout, const nbgl_layoutUpFooter_t *upFooterDesc)
+{
+    nbgl_layoutInternal_t *layoutInt = (nbgl_layoutInternal_t *) layout;
+    layoutObj_t           *obj;
+    nbgl_text_area_t      *textArea;
+    nbgl_line_t           *line;
+    nbgl_button_t         *button;
+
+    LOG_DEBUG(LAYOUT_LOGGER, "nbgl_layoutAddUpFooter():\n");
+    if (layout == NULL) {
+        return -1;
+    }
+    if ((upFooterDesc == NULL) || (upFooterDesc->type >= NB_UP_FOOTER_TYPES)) {
+        return -2;
+    }
+
+    layoutInt->upFooterContainer
+        = (nbgl_container_t *) nbgl_objPoolGet(CONTAINER, layoutInt->layer);
+    layoutInt->upFooterContainer->obj.area.width = SCREEN_WIDTH;
+    layoutInt->upFooterContainer->layout         = VERTICAL;
+    // maximum 4 children for long press button
```

Comment:
4 for long press. Is it the absolute maximum, whatever the types?
What about other type max children?

Reply:
4 is the worst case in the switch below. Then we allocate an array of 4 children, before the switch, and we are sure that it will be suffisiant

Reply:
There is already a comment :` // maximum 4 children for long press button`